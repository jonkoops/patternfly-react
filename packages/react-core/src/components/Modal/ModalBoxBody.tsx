import { HTMLProps, ReactNode, FunctionComponent } from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/ModalBox/modal-box';

export interface ModalBoxBodyProps extends HTMLProps<HTMLDivElement> {
  /** Content rendered inside the modal box body. */
  children?: ReactNode;
  /** Additional classes added to the modal box body. */
  className?: string;
}

export const ModalBoxBody: FunctionComponent<ModalBoxBodyProps> = ({
  children = null,
  className = '',
  ...props
}: ModalBoxBodyProps) => (
  <div {...props} className={css(styles.modalBoxBody, className)}>
    {children}
  </div>
);
ModalBoxBody.displayName = 'ModalBoxBody';
