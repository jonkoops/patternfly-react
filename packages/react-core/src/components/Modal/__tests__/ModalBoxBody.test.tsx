/**
 * @vitest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';
import { expect, test } from 'vitest';

import { ModalBoxBody } from '../ModalBoxBody';

test('ModalBoxBody Test', () => {
  const { asFragment } = render(
    <ModalBoxBody id="id" className="test-box-class">
      This is a ModalBox header
    </ModalBoxBody>
  );
  expect(asFragment()).toMatchSnapshot();
});
