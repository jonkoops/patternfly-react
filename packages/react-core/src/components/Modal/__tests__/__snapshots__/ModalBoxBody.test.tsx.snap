// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModalBoxBody Test 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-modal-box__body test-box-class"
    id="id"
  >
    This is a ModalBox header
  </div>
</DocumentFragment>
`;
